text/0000-cargo-libstd-awareness.md

```diff
@@ -0,0 +1,236 @@
+From the users's perspective, `stdlib = true` simply indicates that the depended-on crate is from the standard library.
+The version for stdlib crates comes from the version of Rust their interfaces are defined in.
+A version requirement must be specified.
+The full breadth of options available with our existing dependencies, e.g. features and overrides, will be supported.
+
+## Implicit dependencies
+
+For backwards compatibility, Cargo must inject such standard library dependencies for existing packages.
+These injected standard library dependencies are called "implicit dependencies" because the user does not specify them explicitly.
+We have an obligation to not break packages depending only on stable interfaces, so the implicit dependencies will include both `std` and `core`:
+```toml
+[dependencies]
+core = { version = "^1.0", stdlib = true }
+std  = { version = "^1.0", stdlib = true }
+
+[dev-dependencies]
+core = { version = "^1.0", stdlib = true }
+std  = { version = "^1.0", stdlib = true }
+test = { version = "^1.0", stdlib = true }
+
+[build-dependencies]
+core = { version = "^1.0", stdlib = true }
+std  = { version = "^1.0", stdlib = true }
+```
+The version requirement for `core` of `^1.0` may seem odd because core was not stable at the time, but anything else would either break newer packages using core, or prevent older packages from working on versions of Rust predating core's stabilization.
+Remember that rustc only complains if an unstable crate is actually imported, so the Cargo dependency on its own is harmless.
+
+`test` is a similar scenario.
+While importing it explicitly remains unstable, it's currently injected and thus needs to be built.
+Other dependencies of `std` besides core we don't need to worry about, because they are only transitive dependencies through `std`, not direct dependencies.
+
+Now, not all crates depend on these crates, so there must be a way to opt out.
+For this, we introduce a new `implicit-dependencies` key.
+It is defined by default as:
+```toml
+implicit-dependencies = ["primary", "build", "dev"]
+```
+This indicates each of `dependencies`, `build-dependencies`, and `dev-dependencies` maps (respectively) is augmented with implicit elements.
+A manual definition may be that or almost any subset, in which case only the included dependency maps are augmented.
+
+Finally, if an (explicit) dependency conflicts with one of the implicit defaults, that category of implicit dependency will be skipped.
+For example, if a crate explicit depends on `std` as a build-dependency, neither `std` nor any other implicit build dependency will be injected.
+This final rule means must packages won't need to use the implicit-dependencies key, because either implicit dependencies will be used, or a `core` or `std` explicit dependency will be present.
+An exception is `core` itself, which must of course not depend on `core` or `std` implicitly or explicitly.
```

Comment:
It looks to me like core is the *only* exception. Can you state that explicitly? And if not, say what other use cases there are.